edxval/api.py

```diff
@@ -230,6 +232,54 @@ def update_video_transcript(video_id, language, transcript_url, transcript_forma
     return TranscriptSerializer(transcript).data
 
 
+def get_3rd_party_transcription_plans():
+    """
+    Retrieves 3rd party transcription plans.
+    """
+    return THIRD_PARTY_TRANSCRIPTION_PLANS
+
+
+def get_transcript_preferences(course_id):
+    """
+    Retrieves course wide transcript preferences
+
+    Arguments:
+        course_id (CourseKey): a course key
+    """
+    transcript_preference = None
+    transcript_preference_set = TranscriptPreference.objects.filter(course_id=course_id)
+    if transcript_preference_set.exists():
+        transcript_preference = TranscriptPreferenceSerializer(transcript_preference_set.first()).data
+
+    return transcript_preference
+
+
+def create_or_update_transcript_preferences(course_id, provider, **kwargs):
+    """
+    Creates or updates course-wide transcript preferences
+
+    Arguments:
+        course_id(CourseKey): a course key
+        provider(str): transcript provider 3play/cielo24
+
+    Keyword Arguments:
+        kwargs(dict): keyword arguments
+    """
+    try:
+        transcript_preference = TranscriptPreference.objects.get(course_id=course_id)
+        transcript_preference.provider = provider
+        transcript_preference.three_play_turnaround = kwargs.get('three_play_turnaround')
+        transcript_preference.cielo24_turnaround = kwargs.get('cielo24_turnaround')
+        transcript_preference.cielo24_fidelity = kwargs.get('cielo24_fidelity')
+        transcript_preference.preferred_languages = kwargs.get('preferred_languages')
```

Comment:
Is it possible to have None or empty value in these kwargs, if yes then how about default values.